packages/core/src/parsers/request.ts

```diff
@@ -0,0 +1,30 @@
+import { IncomingMessage } from 'http';
+import { mimes } from '../constants';
+
+const handleChunks = (contentType: string, chunks: string) => {
+  switch (contentType) {
+    case mimes.JSON:
+      return JSON.parse(chunks);
+    default:
+      return chunks;
+  }
+};
+
+const requestParser = (req: IncomingMessage) => {
+  return new Promise(resolve => {
+    const { method, headers } = req;
+    if (!method || !(method.toLowerCase() in ['post', 'put', 'patch'])) {
+      resolve({});
+      return;
+    }
+    let chunks = '';
+    req.on('data', chunk => {
+      chunks += chunk.toString();
```

Comment:
should probably handle chunk transformation in `handleChunks` since they may not be `toString()`able unless e.g. it's json data